src/fty_srr_dto.cc

```diff
@@ -28,10 +28,136 @@
 
 #include "fty_common_dto_classes.h"
 
+#include <map>
+
 namespace dto 
 {
     namespace srr 
     {
+        /**
+         * Config request object
+         * @param data
+         * @param object
+         */
+        static std::map<Action, std::string> actionInString =
+        {
+            {Action::SAVE,    "save"},
+            {Action::RESET,   "reset"},
+            {Action::RESTORE, "restore"},
+            {Action::UNKNOWN, "unknown"}
+        };
+
+        std::string actionToString(Action action)
+        {
+            try
+            {
+                return actionInString.at(action);
+            }
+            catch(...)
+            {
+                return "unknown";
+            }
```

Comment:
Don't throw an exception merely for flow control. Use `std::map::find()` and check against `std::map::end()`.